selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/FetchAndMergeRemoteBranchIntoLocalTest.java

```diff
@@ -0,0 +1,190 @@
+  private static final String PROJECT_NAME = NameGenerator.generate("FetchAndMergeTest-", 4);
+  private static final String CHANGE_CONTENT =
+      String.format("//change_content-%s", String.valueOf(System.currentTimeMillis()));
+
+  private GitHub gitHub;
+  private GHRepository gitHubRepository;
+
+  @Inject private TestWorkspace ws;
+  @Inject private Ide ide;
+  @Inject private TestUser productUser;
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject
+  @Named("github.password")
+  private String gitHubPassword;
+
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Menu menu;
+  @Inject private Git git;
+  @Inject private Events eventsPanel;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Consoles consoles;
+  @Inject private TestSshServiceClient testSshServiceClient;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private TestGitHubServiceClient gitHubClientService;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    gitHub = GitHub.connectUsingPassword(gitHubUsername, gitHubPassword);
+    gitHubRepository = gitHub.createRepository(REPO_NAME).create();
+    String commitMess = String.format("add-new-content-%s ", System.currentTimeMillis());
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath = Paths.get(getClass().getResource("/projects/guess-project").getPath());
+    gitHubClientService.addContentToRepository(entryPath, commitMess, gitHubRepository);
+    ide.open(ws);
+  }
+
+  @AfterClass
+  public void deleteRepo() throws IOException {
+    gitHubRepository.delete();
+  }
+
+  @Test
+  public void fetchUpdatesAndMergeRemoteBranchTolocal() throws IOException {
+    // preconditions and import the test repo
+    String textFile = "README.md";
+    String javaFile = "AppController";
+    String jspFile = "index.jsp";
+    String pathToJavaFile = "src/main/java/org/eclipse/qa/examples";
+    String pathToJspFile = "src/main/webapp";
+    String originMaster = "origin/master";
+    String mergeMess1 = "Fast-forward Merged commits:";
+    String mergeMess2 = "New HEAD commit: ";
+    String mergeMess3 = "Already up-to-date";
+
+    projectExplorer.waitProjectExplorer();
+    String repoUrl = String.format("https://github.com/%s/%s.git", gitHubUsername, REPO_NAME);
+    git.importJavaApp(repoUrl, PROJECT_NAME, MAVEN);
+
+    // change content in the test repo on GitHub
+    deleteFileOnGithubSide(String.format("%s/%s", pathToJspFile, jspFile), "delete index.jsp");
+    changeContentOnGithubSide(
+        String.format("%s/%s.java", pathToJavaFile, javaFile), CHANGE_CONTENT);
+    changeContentOnGithubSide(textFile, CHANGE_CONTENT);
+
+    performFetch();
+
```

Comment:
How do we know that fetch has been performed as well?

Reply:
Add the check of the message in the console that `git fetch` is preformed well